rust/kcl-lib/src/execution/types.rs

```diff
@@ -1045,6 +1054,26 @@ impl KclValue {
         convert_units: bool,
         exec_state: &mut ExecState,
     ) -> Result<KclValue, CoercionError> {
+        match self {
+            KclValue::Tuple { value, meta } if value.len() == 1 && !matches!(ty, RuntimeType::Tuple(..)) => {
+                if let Ok(coerced) = value[0].coerce(ty, convert_units, exec_state) {
+                    return Ok(KclValue::Tuple {
+                        value: vec![coerced],
+                        meta: meta.clone(),
+                    });
+                }
+            }
+            KclValue::HomArray { value, .. } if value.len() == 1 && !matches!(ty, RuntimeType::Array(..)) => {
+                if let Ok(coerced) = value[0].coerce(ty, convert_units, exec_state) {
+                    return Ok(KclValue::HomArray {
```

Comment:
I would have expected that coercing an array to a non-array would result in a non-array or error. What's the reasoning for this?

Reply:
So this works now? Even though `a` isn't an array?

```
a = 1: [number]
one = a[0]
```

Reply:
I believe that works currently and with this PR

Reply:
This works on main, but on this branch, I get this error.
<img width="507" alt="Screenshot 2025-05-27 at 10 22 58 AM" src="https://github.com/user-attachments/assets/7bb5f12a-4b4b-4fbf-904a-be35ed6c1765" />